source/Nevermore/Advanced/Queryable/QueryTranslator.cs

```diff
@@ -45,131 +45,146 @@ protected override Expression VisitMethodCall(MethodCallExpression node)
                 throw new NotSupportedException();
             }
 
-            if (methodInfo.Name == nameof(NevermoreQueryableExtensions.WhereCustom))
+            switch (methodInfo.Name)
             {
-                Visit(node.Arguments[0]);
-                if (node.Arguments[1] is ConstantExpression { Value: string } constantExpression)
+                case nameof(NevermoreQueryableExtensions.WhereCustom):
                 {
+                    Visit(node.Arguments[0]);
+                    if (node.Arguments[1] is not ConstantExpression { Value: string } constantExpression)
```

Comment:
I inverted the if here, and changed the toplevel thing to a switch, and added ThenBy + ThenByDesecending, but none of the logic for the other statements has changed at all